amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisher.java

```diff
@@ -216,12 +216,15 @@ public void restartFrom(RecordsRetrieved recordsRetrieved) {
         PrefetchRecordsRetrieved prefetchRecordsRetrieved = (PrefetchRecordsRetrieved) recordsRetrieved;
         resetLock.writeLock().lock();
         try {
+            // Reset the demand from ShardConsumer, to prevent this publisher from delivering events to stale RX-Java
+            // Subscriber. Publishing will be unblocked when the demand is communicated by the new Rx-Java subscriber.
+            requestedResponses.set(0);
```

Comment:
this class keep track of so much state, and they all need to be reset at restartFrom call. I'm wondering if pulling out the state tracking variable in one common place, and call reset on that one to start with clean state helps longer term maintainability of this class. What do you think?